apps/judicial-system/backend/src/app/modules/case/guards/caseCompleted.guard.ts

```diff
@@ -12,7 +12,7 @@ import { Case } from '../models'
 
 @Injectable()
 export class CaseCompletedGuard implements CanActivate {
-  async canActivate(context: ExecutionContext): Promise<boolean> {
+  canActivate(context: ExecutionContext): boolean {
```

Comment:
I see that you are removing async in a couple of places but you are not removing any awaits. Was this function never being awaited on?

Reply:
This function is called as part of middleware execution. The framework uses awaits as neede.